### PR TITLE
Ga 108 compare local in prm database access type definitions latest

### PR DESCRIPTION
**Pull request by patelila**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/GA-108

### Change description ###

PRM : Compare the local (held in PRM database) access type definitions with the latest

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/AccessTypesRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @Autowired
+    private AccessTypesRepository accessTypesRepository;
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_access_types.sql"})
+    public void shouldGetAccessTypesEntity() {
+        AccessTypesEntity accessTypes = accessTypesRepository.getAccessTypesEntity();
+
+        assertNotNull(accessTypes);
+        assertNotNull(accessTypes.getVersion());
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/AccessTypesRepositoryIntegrationTest.java]: updated in [d785d12](https://github.com/hmcts/am-org-role-mapping-service/pull/1728/commits/d785d121e0f8e6c097ed8edc4cd57884c9b520cb)

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AccessTypesBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    @MethodSource("modifiedAccessTypeCombinations")
+    void identifyUpdatedOrgProfileIdsWhenOneFieldIsModified(RestructuredAccessTypes modifiedAccessTypes) {
+        RestructuredAccessTypes restructuredAccessTypes =
+                buildRestructuredAccessTypes(true, true, "caseTypeId1", "roleName1",
+                        "groupRole1", "caseGroupIdTemplate1", true);
+
+        List<String> modifiedOrgProfiles =
+                AccessTypesBuilder.identifyUpdatedOrgProfileIds(restructuredAccessTypes, modifiedAccessTypes);
+
+        assertTrue(modifiedOrgProfiles.contains("SOLICITOR_ORG"));
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AccessTypesBuilderTest.java]: updated test with more changes with 3 different org types

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/helper/AccessTypesBuilder.java [hunk 3]
```diff
@@ -3,12 +3,111 @@
...
+        List<OrganisationProfile> organisationProfileList = organisationProfilesMap.keySet().stream()
+                .map(key -> new OrganisationProfile(key, organisationProfilesMap.get(key)))
+                .toList();
+
+        return new RestructuredAccessTypes(organisationProfileList);
+    }
+
+    public static List<String> identifyUpdatedOrgProfileIds(RestructuredAccessTypes restructuredCcdAccessTypes,
+                                                            RestructuredAccessTypes prmStoredAccessTypes) {
+        if (prmStoredAccessTypes.getOrganisationProfiles() == null) {
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/helper/AccessTypesBuilder.java:73]: yes - see test `identifyNoOrgProfileIdsWhenAccessTypesEqualButInDiffOrderTest` which checks no org profiles are identified when elements are the same but in different order

**Review by helenbirdhmcts — APPROVED** [review]
lgtm

**Review by sanjayparekh-hmcts — APPROVED** [review]

**Comment by TomForniCGI** [comment]
Merged into GA-54

**Pull request closed without merging**